src/checksums.ts

```diff
@@ -6,8 +6,11 @@ export async function fetchValidChecksums(
   allowSnapshots: boolean
 ): Promise<string[]> {
   const all = await httpGetJsonArray('https://services.gradle.org/versions/all')
-  const withChecksum = all.filter(entry =>
-    entry.hasOwnProperty('wrapperChecksumUrl')
+  const withChecksum = all.filter(
+    entry =>
+      typeof entry === 'object' &&
+      entry != null &&
+      entry.hasOwnProperty('wrapperChecksumUrl')
   )
```

Comment:
Can you elaborate upon this change? What changed here?

Reply:
The upgrade of TypeScript and eslint required the code to be more static and strict.